src/deepsparse/yolo/pipelines.py

```diff
@@ -245,8 +275,17 @@ def process_engine_outputs(
 
         batch_boxes, batch_scores, batch_labels = [], [], []
 
-        for image_output in batch_output:
-            batch_boxes.append(image_output[:, 0:4].tolist())
+        original_image_shapes = kwargs.get("original_image_shapes")
+        for idx, image_output in enumerate(batch_output):
+            original_image_shape = (
+                original_image_shapes[idx] if idx < len(original_image_shapes) else None
+            )
+            batch_boxes.append(
+                self._scale_boxes(
+                    boxes=image_output[:, 0:4],
+                    original_image_shape=original_image_shape,
```

Comment:
I would still remove as much of boilerplate from this logic here and let _scale_boxes do the logic from lines L278 to 282

Reply:
I don't think that would improve the readability of the code, will leave this as is for now if not a blocker?